src/signTx.c

```diff
@@ -199,7 +199,9 @@ static inline void advanceStage()
 		ctx->stage = SIGN_STAGE_WITNESSES;
 		initTxWitnessCtx();
 
-		break;
+		if (ctx->numWitnesses != 0) {
```

Comment:
```suggestion
		if (ctx->numWitnesses > 0) {
```

Reply:
Not sure why it's better, but if you prefer that, can be done.

Reply:
Mostly just because we do it like this in other places. (probably because human are less good at processing negations?...)